vegbilder-client/src/components/MapContainer/MapContainer.tsx

```diff
@@ -0,0 +1,150 @@
+
+  const createBboxForVisibleMapArea = useCallback(() => {
+    // Add some padding to the bbox because the meridians do not perfectly align with the vertical edge of the screen (projection issues)
+    let paddingX = (east - west) * 0.1;
+    let paddingY = (north - south) * 0.1;
+    if (settings.debugMode) {
+      // Simulate a smaller visible map area for debugging purposes.
+      paddingX = (west - east) * 0.2;
+      paddingY = (south - north) * 0.2;
+    }
+
+    return {
+      south: south - paddingY,
+      west: west - paddingX,
+      north: north + paddingY,
+      east: east + paddingX,
+    };
+  }, [south, west, north, east]);
+
+  const fetchImagePointsFromNewestYearWhereUserClicked = (latlng: ILatlng) => {
+    (async () => {
+      const bboxVisibleMapArea = createBboxForVisibleMapArea();
+      if (isFetching) return;
+      if (
+        !loadedImagePoints ||
+        currentYear === 'Nyeste' ||
+        !isBboxWithinContainingBbox(bboxVisibleMapArea, loadedImagePoints.bbox)
+      ) {
+        availableYears.some(async (year) => {
+          setIsFetching(true);
+          const targetBbox = createSquareBboxAroundPoint(latlng, settings.targetBboxSize);
+          const { imagePoints, expandedBbox } = await getImagePointsInTilesOverlappingBbox(
+            targetBbox,
+            year
+          );
+          if (imagePoints && imagePoints.length > 0) {
+            setLoadedImagePoints({
+              imagePoints: imagePoints,
+              bbox: expandedBbox,
+              year: year,
+            });
+            selectNearestImagePointToClickedCoordinates(imagePoints, latlng);
+            setIsFetching(false);
+            return imagePoints.length > 0;
+          }
+        });
+      }
+    })();
+  };
+
+  const selectNearestImagePointToClickedCoordinates = useCallback(
+    (imagePoints: IImagePoint[], latlng) => {
+      if (!imagePoints || !imagePoints.length || !currentCoordinates) return;
+      const nearestImagePoint = findNearestImagePoint(imagePoints, latlng);
+      if (nearestImagePoint) {
+        const year = nearestImagePoint.properties.AAR;
+        setCurrentImagePoint(nearestImagePoint);
+        setCurrentYear(year);
+        setQueryParamYear(year.toString());
+        showMessage(
+          `Setter årstallet til ${year}, som er det året med de nyeste bildene i området.`
+        );
+      } else {
+        showMessage(
+          'Fant ikke noen bilder i nærheten av der du klikket. Prøv å klikke et annet sted.'
+        );
+      }
+      return nearestImagePoint;
+    },
+    [
+      loadedImagePoints,
+      currentCoordinates,
+      setCurrentImagePoint,
+      setCurrentYear,
+      setQueryParamYear,
+      showMessage,
+    ]
+  );
+
+  /* Fetch image points in new target area when the user clicks on the map. If we find an image, we set the year to the year where we found the image.
+   */
+  const handleClick = (event: any) => {
```

Comment:
Any chance for å få type på event? 😄

Reply:
Antar det er Map-komponenten som legger føringene for denne. Hvis ikke denne komponenten har typer fra før, ser det ut som du kan legge det til med `npm install --save-dev @types/react-leaflet`